File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/character/CharacterListAccountDTO.java
package dev.spiralmoon.maplestory.api.dto.character;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.util.List;

/**
 * 메이플스토리 계정
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterListAccountDTO {

    /**
     * 메이플스토리 계정 식별자
     */
    @SerializedName("account_id")
    private String accountId;

    /**
     * 캐릭터 목록
     */
    @SerializedName("character_list")
    private List<CharacterListAccountCharacterDTO> characterList;
}

File: js/src/maplestory/api/dto/character/characterListAccountDto.ts
import { CharacterListAccountCharacterDto } from './characterListAccountCharacterDto';
import { CharacterListAccountBody } from '../../response/character/characterListBody';

/**
 * 메이플스토리 계정
 */
class CharacterListAccountDto {

  /**
   * 메이플스토리 계정 식별자
   */
  accountId: string;

  /**
   * 캐릭터 목록
   */
  characterList: CharacterListAccountCharacterDto[];

  constructor(obj: CharacterListAccountBody) {
    const { account_id, character_list } = obj;

    this.accountId = account_id;
    this.characterList = character_list.map(character => new CharacterListAccountCharacterDto(character));
  }
}

export { CharacterListAccountDto };
